data_prep/graph_dataset.py

```diff
@@ -1,72 +1,153 @@
 import abc
 
-from data_prep.dataset import Dataset
+import torch
 from torch_geometric.data import Dataset as GeometricDataset
+from torch_geometric.data import Data
+from transformers import RobertaTokenizerFast
+
+from data_prep.dataset import Dataset
+from data_prep.graph_utils import tf_idf_mtx, get_PMI
 
 
 class GraphDataset(Dataset, GeometricDataset):
     """
     Parent class for graph datasets.
     Require to implement a preprocess and generate_features methods.
     """
-    def __init__(self, *args, **kwargs):
-        # preprocess text
-        # compute tf.idf and pmi
-        # generate edges
-        # generate features
-        # build data object
-        raise NotImplementedError
 
+    def __init__(self, corpus, device):
+        self._device = device
+        self._tokenizer = RobertaTokenizerFast.from_pretrained("roberta-base")
```

Comment:
We only need the tokenizer for the RobertaGNNDataset class right? Why is it initialized here?

Reply:
No, the `PureGraphDataset` also uses it to know what are the words of the Roberta vocabulary